cli.js

```diff
@@ -36,117 +33,64 @@ const cli = meow(`
 	string: [
 		'plugin',
 		'optimizationLevel',
+		'outdir',
 		'install'
 	],
 	alias: {
+		O: 'optimizationLevel',
 		P: 'plugin',
 		i: 'interlaced',
-		o: 'optimizationLevel',
+		o: 'outdir',
 		p: 'progressive'
 	}
 });
 
-function isFile(path) {
-	if (/^[^\s]+\.\w*$/.test(path)) {
-		return true;
-	}
-
-	try {
-		return fs.statSync(path).isFile();
-	} catch (err) {
-		return false;
-	}
-}
-
 const DEFAULT_PLUGINS = [
 	'gifsicle',
-	'jpegtran',
+	'mozjpeg',
 	'optipng',
 	'svgo'
 ];
 
-function run(src, dest) {
-	const plugins = cli.flags.plugin ? arrify(cli.flags.plugin) : DEFAULT_PLUGINS;
-	const imagemin = new Imagemin().src(src);
+const requirePlugins = (plugins, opts) => plugins.map(x => {
+	try {
+		x = require(`imagemin-${x}`)(opts);
+	} catch (err) {
+		console.error(stripIndent(`
+			Unknown plugin: ${x}
 
-	plugins.forEach(name => {
-		let plugin;
+			Maybe you forgot to install the plugin?
+			You can install it with:
 
-		switch (name) {
-			case 'gifsicle':
-			case 'jpegtran':
-			case 'optipng':
-				plugin = Imagemin[name](cli.flags);
-				break;
-			case 'svgo':
-				plugin = Imagemin.svgo();
-				break;
-			default:
-				try {
-					plugin = require('imagemin-' + name)(cli.flags);
-				} catch (err) {
-					console.error([
-						'Unknown plugin ' + name,
-						'',
-						'Maybe you forgot to install the plugin?',
-						'You can install it with:',
-						'  $ npm install -g imagemin-' + name
-					].join('\n'));
-					process.exit(1);
-				}
-				break;
-		}
+			  $ npm install -g imagemin-${x}
+		`).trim());
+		process.exit(1);
+	}
 
-		imagemin.use(plugin);
-	});
+	return x;
+});
 
-	if (dest) {
-		imagemin.dest(dest);
-	}
+const run = (input, opts) => {
+	opts.plugin = opts.plugin ? arrify(opts.plugin) : DEFAULT_PLUGINS;
```

Comment:
Don't modify the user-supplied object. This will cause havok if the users uses the same object for multiple calls.